src/utils/fetch.js

```diff
@@ -0,0 +1,224 @@
+
+            if (response.status < 400) {
+                if (request.dataType === 'json' || request.headers.accept === 'application/json') {
+                    return response.json();
+                } else if (request.dataType === 'text' || (response.headers.get('Content-Type') || '').toLowerCase().indexOf('text/') === 0) {
+                    return response.text();
+                } else {
+                    return response;
+                }
+            } else {
+                onFetchFail(this, request.url, response);
+                return Promise.reject(response);
+            }
+        }).catch((error) => {
+            onFetchFail(this, request.url, {});
+            throw error;
+        });
+    }
+
+    return this.fetchWithFailover(request, true);
+}
+
+/**
+ *
+ * @param url
+ * @param options
+ * @param timeoutMs
+ * @returns {Promise<unknown>}
+ */
+export function fetchWithTimeout(url, options, timeoutMs) {
+    return new Promise((resolve, reject) => {
+        const timeout = setTimeout(reject, timeoutMs);
+
+        options = options || {};
+        options.credentials = 'same-origin';
+
+        fetch(url, options).then((response) => {
+            clearTimeout(timeout);
+            resolve(response);
+        }).catch((error) => {
+            clearTimeout(timeout);
+            reject(error);
+        });
+    });
+}
+
+export function getTryConnectPromise(instance, url, state, resolve, reject) {
+    fetchWithTimeout(instance.getUrl('system/info/public', null, url), {
+        method: 'GET',
+        accept: 'application/json'
+    }, 15000).then(() => {
+        if (!state.resolved) {
+            state.resolved = true;
+
+            console.log("Reconnect succeeded to " + url);
+            instance.serverAddress(url);
+            resolve();
+        }
+    }).catch(() => {
+        if (!state.resolved) {
+            console.log("Reconnect failed to " + url);
+
+            state.rejects++;
+            if (state.rejects >= state.numAddresses) {
+                reject();
+            }
+        }
+    });
+}
+
+function tryReconnectInternal(instance) {
+    const addresses = [];
+    const addressesStrings = [];
+
+    const serverInfo = instance.serverInfo();
+    if (serverInfo.LocalAddress && addressesStrings.indexOf(serverInfo.LocalAddress) === -1) {
+        addresses.push({ url: serverInfo.LocalAddress, timeout: 0 });
+        addressesStrings.push(addresses[addresses.length - 1].url);
+    }
+    if (serverInfo.ManualAddress && addressesStrings.indexOf(serverInfo.ManualAddress) === -1) {
+        addresses.push({ url: serverInfo.ManualAddress, timeout: 100 });
+        addressesStrings.push(addresses[addresses.length - 1].url);
+    }
+    if (serverInfo.RemoteAddress && addressesStrings.indexOf(serverInfo.RemoteAddress) === -1) {
+        addresses.push({ url: serverInfo.RemoteAddress, timeout: 200 });
+        addressesStrings.push(addresses[addresses.length - 1].url);
+    }
```

Comment:
We also definitely removed these from the server, although I'm not sure about the API client.